File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/erlang_solutions/3277-maximum-xor-score-subarray-queries.erl
-module(solution).
-export([max_xor_subarray/2]).

max_xor_subarray(Nums, Queries) ->
    Prefix = lists:foldl(fun(X, Acc) -> [hd(Acc) bxor X | Acc] end, [0], Nums),
    PrefixList = lists:reverse(Prefix),
    lists:map(fun({L, R}) -> 
        lists:max([lists:nth(L + 1, PrefixList) bxor lists:nth(R + 1, PrefixList)])
    end, Queries).

File: MATTHEW/leetcode/AIcoding/5.Qwen-3/rust_solutions/3277-maximum-xor-score-subarray-queries.rs
impl Solution {

use std::cmp::max;

struct Solution {}

impl Solution {
    pub fn max_xor_subarray(nums: Vec<i32>, queries: Vec<Vec<i32>>) -> Vec<i32> {
        let n = nums.len();
        let mut res = Vec::new();

        for q in queries {
            let l = q[0] as usize;
            let r = q[1] as usize;
            let mut max_xor = 0;
            let mut current = 0;

            for i in l..=r {
                current ^= nums[i];
                max_xor = max(max_xor, current);
            }

            res.push(max_xor);
        }

        res
    }
}
}